hyper/http20/stream.py

```diff
@@ -122,8 +122,17 @@ def file_iterator(fobj):
             chunks = (data[i:i+MAX_CHUNK]
                       for i in range(0, len(data), MAX_CHUNK))
 
-        for chunk in chunks:
-            self._send_chunk(chunk, final)
+        # since we need to know when we have a last package we need to know
+        # if there is another package in advance
+        cur_chunk = next(chunks)
```

Comment:
This will explode if there are no chunks to send: I think you need to write a test case for that and refactor.

Reply:
Done